### PR TITLE
Implement proxies for Azure exporters

### PR DESCRIPTION
**Pull request by lzchen**
Add the ability to specify proxies that the request to Azure Monitor backend will pass through.

Users will pass in a string representing a dict format.

See [requests](https://requests.readthedocs.io/en/master/user/advanced/#proxies) for more details.

### CODE REVIEW

File: azure_monitor/src/azure_monitor/options.py [hunk 1]
```diff
@@ -103,6 +108,10 @@ def _initialize(self) -> None:
         )
         self.endpoint = endpoint + "/v2/track"
 
+        # proxies
+        if self.proxies is None:
+            self.proxies = "{}"
```
Comment (hectorhdzg) [inline azure_monitor/src/azure_monitor/options.py]: is not supposed to be typing.Dict[str, str]?

File: azure_monitor/src/azure_monitor/options.py [hunk 2]
```diff
@@ -49,6 +51,7 @@ def __init__(
         self,
         connection_string: str = None,
         instrumentation_key: str = None,
+        proxies: typing.Dict[str, str] = None,
```
Comment (hectorhdzg) [inline azure_monitor/src/azure_monitor/options.py:55]: Maybe setting the default as {} instead of None to avoid check afterwards

Comment (lzchen) [reply azure_monitor/src/azure_monitor/options.py:55]: Mutable defaults are a common pitfall in Python. https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

File: azure_monitor/src/azure_monitor/options.py (hunk 1 above)
Comment (lzchen) [reply azure_monitor/src/azure_monitor/options.py]: Fixed

**Review by hectorhdzg — APPROVED** [review]

**Pull request merged**